scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/SessionCache.java

```diff
@@ -12,14 +12,17 @@
  * 
  * Contributors:
  *    Bosch Software Innovations GmbH - Initial creation
+ *    Achim Kraus (Bosch Software Innovations GmbH) - add javadoc about required
+ *                                                    thread-safe implementation
  ******************************************************************************/
 package org.eclipse.californium.scandium.dtls;
 
 /**
  * A second level cache for current connection state of DTLS sessions.
  * <p>
- * Connection state can be put to the cache and later retrieved by the
- * DTLS session's ID.
+ * Connection state can be put to the cache and later retrieved by the DTLS
+ * session's ID. Though the cache is intended to be used concurrently, it must
```

Comment:
IMHO this should read: `The provided cache is required to be thread-safe because it will be accessed concurrently.`